### ISSUE TITLE
Aligning code blocks

### ISSUE THREAD

**Issue by holocronweaver** [anki-editor/anki-editor#30]
I notice code blocks in published notes are not left-aligned. Instead they follow the default center-alignment. This leads to almost unreadable code until I manually set left-alignment in Anki's editor.

I'm uncertain what the right solution is since there are so many layers involved - should I customize code block alignment via a config in anki-editor, htmlize, org export, Anki card type, or something else?

**Comment by shunlog** [comment]
I've noticed that centered code is awkward, but my code snippets weren't big enough for this to be too bothering.

Off the bat, I'd say this issue has to be fixed directly in Anki's CSS, just like they fixed [the alignment of list items](https://github.com/ankitects/anki/issues/1801). I'm actually surprised nobody has yet raised an issue for this. I mean, look at how ugly this is
![image](https://github.com/orgtre/anki-editor/assets/48106195/09c65a50-9aff-4732-bbc6-94662c1ccadc)

@holocronweaver when exporting with `anki-editor`, is there syntax coloring? If so, mind sharing a screenshot?

**Comment by shunlog** [comment]
I've just installed anki-editor myself, and discovered the hack @louietan used [since the very beginnings](url) of anki-editor:
```org

    #+BEGIN_EXPORT html
    <div align="left">
    #+END_EXPORT

    #+BEGIN_SRC emacs-lisp
      ...
    #+END_SRC

    #+BEGIN_EXPORT html
    </div>
    #+END_EXPORT
```

Which aligns the code as expected:
![image](https://github.com/orgtre/anki-editor/assets/48106195/b4ec135a-3e8e-4ada-af69-c26f77044f71)

Now I'm even more perplexed as to why has nobody created an issue on [ankitects/anki](https://github.com/ankitects/anki) yet.

As for the hack, it's too cumbersome. I'd rather raise an issue on anki myself than use it.

**Comment by shunlog** [comment]
Alright, I've created an issue on Anki https://github.com/ankitects/anki/issues/2573 to align the `<pre>` elements to the left. org-mode wraps the code in this element on export, so it should work afterwards. 

As a workaround for now, you can [add this style](https://docs.ankiweb.net/templates/styling.html) to your notes:
```css
pre {
    text-align: left;
}
```

@louietan used this hack in [examples.org](https://github.com/orgtre/anki-editor/blob/master/examples.org):
```org
  #+BEGIN_EXPORT html
  <div align="left">
  #+END_EXPORT

  #+BEGIN_SRC
    ...
  #+END_SRC

  #+BEGIN_EXPORT html
  </div>
  #+END_EXPORT
```

It works but is kinda cumbersome, so we should remember to fix it if Anki updates

**Comment by holocronweaver** [comment]
@shunlog Thanks for your work here! Looks like your pull request was merged, does that mean an upcoming version of Anki will have left alignment for code blocks work out of the box?

**Comment by shunlog** [comment]
@holocronweaver yeah, that's the idea. Guess this issue can be closed now.

**Comment by holocronweaver** [comment]
Thanks! Will close once the fix has been released and I've had a chance to test it.

**Comment by slotThe** [comment]
This is definitely included in Anki 24.04, so I'm closing this.
